src/lib/stack-client.js

```diff
@@ -353,13 +353,25 @@ const getSettingsAppURL = function() {
  * @returns {Promise}
  */
 const init = function({ cozyClient: client, onCreate, onDelete }) {
-  cozyClient = client
-  if (!cozyClient.isLogged) return
-  initializeRealtime({
-    getApp,
-    onCreate,
-    onDelete,
-    cozyClient
+  cozyClient = client // Setting global variable
+  let terminateRealtime
+  const onLogin = () => {
+    terminateRealtime = initializeRealtime({
+      getApp,
+      onCreate,
+      onDelete,
+      cozyClient
+    })
+  }
+  if (cozyClient.isLogged) {
+    onLogin()
+  } else {
+    cozyClient.once('login', onLogin)
```

Comment:
What is the main reason to have only a once handler? Why not just a regular on('login') ? What happens if this client is not logged then login then logout then login back?

Reply:
Fair point. In Banks, we init the bar on `login` from the client. Therefore, the `init` function above is called at each `login`. This means that in banks, the `else` codepath would not be used. I think I am going to remove the commits introducing the once "login". Let's settle on the fact that the bar should be initialized when the client is logged in. WDYT ?

Reply:
In Drive we manually call Bar.init() when it's needed. We display a few onboarding screens after the login, but we don't want to init the bar yet. 

I think this is the responsibility of the App to init the bar when it needs. 

> Let's settle on the fact that the bar should be initialized when the client is logged in.  

Yep.

Reply:
FYI, I removed the `once('login', initRealtime)` commit.